core/shared/src/main/scala/com/monovore/decline/Argument.scala

```diff
@@ -36,6 +37,29 @@ object Argument extends PlatformArguments {
 
   def apply[A](implicit argument: Argument[A]): Argument[A] = argument
 
+  /**
+   * We can't add methods to traits in 2.11 without breaking binary compatibility
+   */
+  implicit final class ArgumentMethods[A](private val self: Argument[A]) extends AnyVal {
+    def map[B](fn: A => B): Argument[B] =
+      new Argument[B] {
+        def read(string: String): ValidatedNel[String, B] =
+          self.read(string).map(fn)
+
+        def defaultMetavar = self.defaultMetavar
+      }
+
+    def mapValidated[B](function: A => ValidatedNel[String, B]): Argument[B] =
+      new Argument[B] {
+        def read(string: String): ValidatedNel[String, B] =
+          self
+            .read(string)
+            .andThen(function)
+
+        def defaultMetavar = self.defaultMetavar
```

Comment:
One thing I worry about with these is preserving the metavar when it _should_ have changed. Let's say you do something like `Argument[String].mapValidated(parseInt)` or similar... you now have an `Argument` that only accepts ints but still has metavar `string`.

I'd be inclined to make this method take a new metavar alongside the function... WDYT? `map` seems safe because it can't reject anything, so from the CLI's perspective the range of valid inputs hasn't changed.

Reply:
I agree that mapValidated keeping the defaultMetavar is a bit weird.

What I did here was add a `def from` method more compactly create Argument instances. I think that handles what I want from mapValidated anyway: I can just call read inside a function that I pass to `def from`.